pkg/common/scheduler/statefulset/scheduler.go

```diff
@@ -181,32 +222,57 @@ func (s *StatefulSetScheduler) scheduleVPod(vpod scheduler.VPod) ([]duckv1alpha1
 		return placements, nil
 	}
 
-	// Need less => scale down
-	if tr > vpod.GetVReplicas() {
-		logger.Infow("scaling down", zap.Int32("vreplicas", tr), zap.Int32("new vreplicas", vpod.GetVReplicas()))
-		if state.schedulerPolicy == EVENSPREAD {
-			//spreadVal is the minimum number of replicas to be left behind in each zone for high availability
-			spreadVal = int32(math.Floor(float64(vpod.GetVReplicas()) / float64(state.numZones)))
-			logger.Infow("number of replicas per zone", zap.Int32("spreadVal", spreadVal))
-			placements = s.removeReplicasEvenSpread(tr-vpod.GetVReplicas(), placements, spreadVal)
+	if state.SchedulerPolicy != "" {
+		// Need less => scale down
+		if tr > vpod.GetVReplicas() {
+			logger.Infow("scaling down", zap.Int32("vreplicas", tr), zap.Int32("new vreplicas", vpod.GetVReplicas()))
+			if state.SchedulerPolicy == scheduler.EVENSPREAD || state.SchedulerPolicy == scheduler.EVENSPREAD_BYNODE {
+				//spreadVal is the minimum number of replicas to be left behind in each failure domain for high availability
+				if state.SchedulerPolicy == scheduler.EVENSPREAD {
+					spreadVal = int32(math.Floor(float64(vpod.GetVReplicas()) / float64(state.NumZones)))
+				} else if state.SchedulerPolicy == scheduler.EVENSPREAD_BYNODE {
+					spreadVal = int32(math.Floor(float64(vpod.GetVReplicas()) / float64(state.NumNodes)))
+				}
+				logger.Infow("number of replicas per domain", zap.Int32("spreadVal", spreadVal))
+				placements = s.removeReplicasEvenSpread(state, tr-vpod.GetVReplicas(), placements, spreadVal)
+			} else {
+				placements = s.removeReplicas(tr-vpod.GetVReplicas(), placements)
+			}
+
+			// Do not trigger the autoscaler to avoid unnecessary churn
+
+			return placements, nil
+		}
+
+		// Need more => scale up
+		logger.Infow("scaling up", zap.Int32("vreplicas", tr), zap.Int32("new vreplicas", vpod.GetVReplicas()))
+		if state.SchedulerPolicy == scheduler.EVENSPREAD || state.SchedulerPolicy == scheduler.EVENSPREAD_BYNODE {
+			//spreadVal is the maximum number of replicas to be placed in each failure domain for high availability
+			if state.SchedulerPolicy == scheduler.EVENSPREAD {
+				spreadVal = int32(math.Ceil(float64(vpod.GetVReplicas()) / float64(state.NumZones)))
+			} else if state.SchedulerPolicy == scheduler.EVENSPREAD_BYNODE {
+				spreadVal = int32(math.Ceil(float64(vpod.GetVReplicas()) / float64(state.NumNodes)))
+			}
+			logger.Infow("number of replicas per domain", zap.Int32("spreadVal", spreadVal))
+			placements, left = s.addReplicasEvenSpread(state, vpod.GetVReplicas()-tr, placements, spreadVal)
 		} else {
-			placements = s.removeReplicas(tr-vpod.GetVReplicas(), placements)
+			placements, left = s.addReplicas(state, vpod.GetVReplicas()-tr, placements)
 		}
 
-		// Do not trigger the autoscaler to avoid unnecessary churn
+	} else if s.policy != nil { //Predicates and priorities must be used for scheduling
+		if errs := ValidatePolicy(s.policy); errs != nil {
```

Comment:
Can we find another place to validate policies? Seems wasteful to do this here.

Reply:
Ok - yeah, could move it to be done right after init within [here](https://github.com/knative-sandbox/eventing-kafka/pull/768/files#diff-58d2cad92561a8118d077178816fb6d2689b58a9ea043ba65969c94077913ebfR216)